src/mint.rs

```diff
@@ -796,175 +728,148 @@ mod tests {
             Ok((transaction, transaction_sigs)) => {
-                assert_eq!(genesis_amount, output_amount);
-                assert_eq!(number_of_fuzzed_output_parents, 0);
+                assert_eq!(genesis_amount, output_total_amount);
+                assert_eq!(dbcs_with_fuzzed_parents.len(), 0);
                 assert!(
                     input_amounts.is_empty()
-                        || inputs_to_create_invalid_owner_proofs
-                            .intersection(&BTreeSet::from_iter(owners.keys().copied()))
+                        || BTreeSet::from_iter(dbcs_with_invalid_ownership_proofs.keys())
+                            .intersection(&BTreeSet::from_iter(owners.keys()))
                             .next()
                             .is_none()
                 );
-                assert!(BTreeSet::from_iter(owners.keys().copied())
-                    .is_subset(&inputs_to_create_owner_proofs));
-
-                // The output amounts should correspond to the output_amounts
-                assert_eq!(
-                    BTreeSet::from_iter(
-                        outputs
-                            .iter()
-                            .map(|o| { DbcHelper::decrypt_amount(&outputs_owner, o) })
-                            .collect::<Result<Vec<_>, _>>()?
-                    ),
-                    BTreeSet::from_iter(output_amounts.into_iter().map(|(_, a)| a))
+                assert!(
+                    BTreeSet::from_iter(owners.keys()).is_subset(&BTreeSet::from_iter(
+                        dbcs_with_valid_ownership_proofs.keys()
+                    ))
                 );
 
-                // The outputs should have been uniquely number from 0 to N (N = # of outputs)
+                // The output amounts should correspond to the output_amounts
                 assert_eq!(
-                    BTreeSet::from_iter(
-                        outputs.iter().map(|content| content.output_number as usize)
-                    ),
-                    BTreeSet::from_iter(0..outputs.len())
+                    BTreeSet::from_iter(dbc_output_amounts),
+                    BTreeSet::from_iter(output_amounts)
                 );
 
                 let (mint_key_set, mint_sig_share) = transaction_sigs.values().next().unwrap();
                 let mint_sig =
                     mint_key_set.combine_signatures(vec![mint_sig_share.threshold_crypto()])?;
 
-                let output_dbcs = Vec::from_iter(outputs.into_iter().map(|content| {
-                    Dbc {
-                        content,
-                        transaction: transaction.clone(),
-                        transaction_sigs: BTreeMap::from_iter(
-                            transaction_sigs
-                                .iter()
-                                .map(|(input, _)| (*input, (genesis_key, mint_sig.clone()))),
-                        ),
-                    }
-                }));
+                let output_dbcs =
```

Comment:
Would it make sense to have a DBC builder as well?

Reply:
Yes, might be nice to have reissue return Dbc shares as we discussed in meeting.